### DISCUSSION TITLE
Sharable Collection configs

### DISCUSSION THREAD

**Discussion by BotiKis** [Icon-Shelf/icon-shelf#71] [Ideas]
This is for *far future* once #64 and #54 are there.

### As a designer
As a designer I want to share a collection and it's metadata with my team.
As a designer I create the icons, set up the collection with subcollections and add tags to the icons.
Now I want to share this setup with my devs.

### As a dev
As a dev I wan't a simple experience so I can focus on coding. I don't want to toy around with creating collections and tagging and use the app for productivitiy. I want open a config and it should magically work.

## Possible implementation
Maybe a config file I can export and ship with the icons folder in git (or a zipfile, whatever).

**Comment by MrRobz** [comment]
Thanks for the idea @BotiKis.
Expanding IconShelf to designers is something I'm also interested in.

Could you answer a few questions about designer-> dev icon workflow in your case:
1) Can you give me some examples of tags you might create for icons.
2) Do devs ask for just one icon at a time or the whole collection.

**Comment by BotiKis** [comment]
Of course - glad to help.

# 1 Tags
I would assign tags for multiple reasons:

## Allows synonyms 
Certain things are often called in different languages in multiple ways. A person maybe wouldnt' try to find an icon under the same name as another. By allowing to tag an icon with multiple tags can improve this a lot.

#### Example
I call an icon `ic_preferences` which displays a wrench 🔧.
Somebody else sees it in the designs and searches for `Settings` and won't find it.
Another person might search for `Wrench` bc that's what it is.

This can be avoided by tagging the icon with synonyms.

## Grouping
Besides subdirectories that will be very useful to structure a set, providing custom tags in addition also helps.
Especially with a big design system streching over multiple products and projects.

#### Example
Imaging have a structure like:
- Website
- Application A
- Application B
- Print

Now we are creating a new project called *marketplace* which is mainly on the _website_ but is integrated in _Application A_ and _Print_ too. I would put the icons in *website* but tag them also with `marketplace` so it's easier to find in bulk.

## Experience and what we done
This is just my experience and maybe inspiration for you.
In my company we have a similar issue organising +10k assets for a graphic design application. We decided to go full in with tags and introduced *scopes* to our system. Scopes are just keywords separated by colons `:` in a tag. There are normal tags but also predefined *scope* we are using for categorisation.

#### Example
We introduced the scope *category*. Any item can be tagged with multiple categories and appear there in our application.
`category:foo:bar, category:foo:lul, category:color:blues:turquoise, category:vehicles:cars`
An item tagged like this will appear in the UI under:
```
foo
  |- bar
  |- lul
color
  |- color
    |- blues
      |- turquise
vehicles
  |- cars
```

# 2 one icon at a time or the whole collection
Really depends on the individual dev. Both happens.
I've used to send a dev the whole current pack when they requested an Icon, hoping they all updated all icons but that totally didn't work out.

They are just focused on their single task and don't care about others. (Not blaming here anyone).
Also they ain't talking to each other and adding the same icon multiple times into the project which makes updating an icon annoyingly tedious.

As a solution I've created a git repo with all icons which I keep up to date.
If anyone from the team (devs, designers, marketing) needs any ressources they find it there.
It is prohibited to rename icons so they must use the original names from git.

Unfortunately git is awful for finding icons visually or by a search. Best way is to clone the repo and use the search function of *Finder* oder the *File explorer*.

Because that ain't nice either, i searched for a tool and found **Icon-Shelf** ;).

---

Sorry for the long post 🥔
